blogposts/2021/dev-tool-time-paul-jolly.md

```diff
@@ -0,0 +1,86 @@
+
+## CUE: A gentle introduction
+
+Paul is a core contributor to the [CUE project](https://github.com/cuelang/cue) and his passion for it enchanted the audience. He spent the bulk of this episode showing us some of what it could do.
+
+In a future post, we’ll share a tutorial that explains more of what CUE can do. For now, consider this a quick tour of some of Paul’s favorite features.
+
+### What is CUE?
+
+CUE, in Paul’s words, “is an open source language for defining, generating and validating all kinds of data,” including configuration, APIs, database schemas, and code.
+
+[Marcel van Lohuizen](https://twitter.com/mpvl_), an [ex-Googler](https://about.sourcegraph.com/blog/ex-googler-guide-dev-tools/), started CUE in 2018. Marcel, Paul said, has a deep background in configuration management, having been part of the [initial Borg team](https://kubernetes.io/blog/2015/04/borg-predecessor-to-kubernetes/) (Borg is a spiritual predecessor to Kubernetes).
+
+CUE solves a number of pain points, such as dealing with white space in YAML or configuration templating. “If you ever find yourself fighting with white space in YAML,” Paul said, “that is an area that CUE can solve.” Essentially, CUE is trying to solve the problem of configurations at scale. Considering how often non-engineers are in charge of maintaining configuration, Paul explained, CUE is also designed to be approachable and accessible.
+
+The end result is the ability to have a “unified view of your configuration across many different systems, as opposed to different configurations and different tooling for different parts of whatever your system may be.”
+
+### What can CUE do?
+
+In one example, Paul showed how, with CUE, you can have data and schema sitting alongside each other. You can also easily translate between different data and schema encodings. CUE can translate between different formats and create a single source of truth that works nicely: “All of your tooling can then hang off that,” Paul said.
+
+In another example, Paul used CUE to validate a YAML file to ensure it obeys the schema. Validation is powerful when you’re trying to “move as early as possible into your configuration pipeline,” he explained. Instead of leaving configuration to the last minute or trying to quickly work it out when your configuration fails, CUE enables you to figure it out early.
+
+In a third example, Paul used a fork of Sourcegraph to show a base directory of the YAML specification of all the Kubernetes objects. Paul noted the different services that form part of the Sourcegraph configuration, explaining how maintaining all the regularity between different services and objects can become a headache. Paul showed how you can use CUE to “view things as a unified configuration.” With CUE, Paul was able to remove boilerplate and move things that are common between configurations to the top.
```

Comment:
We should confirm, but I think it was this deploy repo (https://github.com/sourcegraph/deploy-sourcegraph) instead of a fork of Sourcegraph.

Reply:
Checked and you are correct! Fixed